File: reverse-integer/reverse-integer.py
def reverse(self, x):
    symbol = False
    if x < 0:
        symbol = True
    num = str(abs(x))
    reverseNum = int(num[::-1])
    if symbol:
        reverseNum = -reverseNum
    maxNum = pow(2, 31) - 1
    minNum = -pow(2, 31)
    if reverseNum >= maxNum or reverseNum <= minNum:
        return 0
    return reverseNum

File: reverse-integer/reverse-integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function (x) {
  const sign = x < 0
  const num = Math.abs(x).toString()
  const numList = num.split('')
  let ans = parseInt((numList.reverse()).join(''))
  if (sign) {
    ans = -ans
  }
  const maxNum = Math.pow(2, 31) - 1
  const minNum = -Math.pow(2, 31)
  if (ans >= maxNum || ans <= minNum) {
    return 0
  }
  return ans
};